### ISSUE TITLE
Inverted attribute order

### ISSUE THREAD

**Issue by jaraco** [jaraco/tempora#39]
In #36, we learned that there's an inverted attribute order in:

https://github.com/jaraco/tempora/blob/6579e3cc7ea90ccf634f618e44affca369d33eaf/tempora/schedule.py#L121

@Avasam can you elaborate on the concern?

**Comment by Avasam** [comment]
Sure! Take the following example:
```
class A:
    def method(self, message: str, count: int) -> None:
        print("Message:" + message, count + 1)


class B(A):
    def method(self, count: int, message: str) -> None:
        print("Subclass message:" + message, count + 1)


def do_something(a: A) -> None:
    if isinstance(a, A):
        a.method("High", 5)


# Passes type-checking and isinstance check
do_something(A())

# Passes type-checking and isinstance check, but raises TypeError
do_something(B())
```

**Comment by jaraco** [comment]
Thanks for the thorough explanation. Looking at these functions, I mean for them to be alternate constructors. I don't intend for them to be used as instance methods or for Liskov substitution to apply. I merely wish for `PeriodicCommand` and `DelayedCommand` to be constructable from a specific time and `PeriodicCommandFixedDelay` to be constructed from a time and delay. I would not expect a user to have a `DelayedCommand` class or subclass and be able to construct an instance without knowing which subclass.

The reason I put `target` at the end was because I want the caller to focus first on "when" (`at`, maybe `delay`) and then "what" (target). Same for `DelayedCommand.after`. Putting `target` between `at` and `delay` is awkward from that vantage.

Perhaps there's a better way, perhaps to have a `When` class that encapsulates "at, maybe delay", and then every class could have a consistent interface. Or maybe we could make `target` keyword-only. Or perhaps we factor out `target` as a parameter altogether and add a `do` helper:

```
cmd = DelayedCommand.after(5).do(the_thing)
```

i.e.

```
class DelayedCommand(datetime.datetime):
    @classmethod
    def after(cls, delay) -> Self:
        ...

    def do(self, target) -> Self:
        self.target = target
        return self
```

That still doesn't satisfy LSP, however, as `PeriodicCommandFixedDelay` still requires different inputs to construct.

In short, because these are constructors and not meant to be used as instance methods, LSP doesn't apply.

**Comment by Avasam** [comment]
In that case, a suppression comment linking back here should be fine :)
